src/wp-admin/edit-tags.php

```diff
@@ -38,6 +38,7 @@
  */
 global $post_type;
 
+/** @var WP_Terms_List_Table $wp_list_table */
```

Comment:
Our list tables aren't LSP compatible, so any code editor or IDE that provides intellisense (in addition to PHPStan) raises errors further down when accessing methods or properties that only exist on certain `WP_List_Table` sub-classes. These inline type declarations avoid those errors.

Reply:
Which, in my humble opinion, is doing things the wrong way around.

The LSP violations should be fixed and @hellofromtonya and me have been discussing the problem with those in our live stream sessions recently as well.

Adding comments to avoid addressing the *real* problem, especially comment which are only recognized by a select set of IDEs/tooling, just doesn't make any sense whatsoever to me.

Reply:
That is indeed the preferred and correct way to address the problem, but realistically can that be fixed? And can it be enforced in future list tables? These classes are a free-for-all due to the dynamic method names used by columns and the arbitrary methods introduced in sub-classes.

There's nothing wrong with specifying the type of a variable in a PHPDoc where it cannot be determined statically. In this case it has the same effect as directly instantiating the concrete class instead of instantiating it via the `_get_list_table()` wrapper (which is little more than a non-standard implementation of an autoloader).